public/modules/custom/helfi_rekry_content/helfi_rekry_content.module

```diff
@@ -69,6 +70,67 @@ function _helfi_rekry_content_get_media_image(string|NULL $fid = NULL): ?string
   return NULL;
 }
 
+/**
+ * Get media entity by file id.
+ *
+ * @param string|null $url
+ *   The video url
+ *
+ * @return string|null
+ *   Valid video url or null
+ */
+function _helfi_rekry_check_video_url(string|NULL $url = NULL): ?string {
+  if (!$url) {
+    return NULL;
+  }
+
+  if (str_contains($url, 'youtube') || str_contains($url, 'youtu.be') || str_contains($url, 'helsinkikanava')) {
```

Comment:
Instead of this, we could do something like this: 

```
  try {
    $resolver = \Drupal::service('media.oembed.url_resolver');
    $validate = $resolver->getProviderByUrl($url);
    return $url;
  } catch (\throwable $e) {
    \Drupal::logger('helfi_rekry_content')->notice('Video embed url "' . $url . '"failed validation with message: ' . $e->getMessage());
    return NULL;
  }
```